### PR TITLE
[ML-7986] Update tensorflow to 1.14.0

### PR DESCRIPTION
**Pull request by lu-wang-dl**
- Update tensorflow version to 1.14.0 in `environment.yml`, `project/Dependencies.scala`, and `python/requirements.txt`
- Auto update `*.proto` with the script. All of this type update comes from tensorflow.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -81,7 +81,13 @@ object Shading extends Build {
       ShadeRule.rename("com.google.protobuf.**" -> "org.tensorframes.protobuf3shade.@1").inAll,
       ShadeRule.rename("google.protobuf.**" -> "org.tensorframes.google.protobuf3shade.@1").inAll
     ),
-    assemblyOption in assembly := (assemblyOption in assembly).value.copy(includeScala = false)
+    assemblyOption in assembly := (assemblyOption in assembly).value.copy(includeScala = false),
+    assemblyMergeStrategy in assembly := {
+        case PathList("org", "tensorflow", xs @ _*) => MergeStrategy.first
```
Comment (mengxr) [inline project/Build.scala]: Please add an inline comment. This happens to fix the issue, but it is not guaranteed by contract. If we can explicitly remove the class from libtensorflow_jni jar, that is safer (not needed for now).

File: src/main/scala/org/tensorflow/NativeLibrary.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (mengxr) [inline src/main/scala/org/tensorflow/NativeLibrary.java]: Link to the issue.

**Comment by mengxr** [comment]
@lu-wang-dl Could you run build/test several times to confirm the `first` behavior is stable?

**Comment by lu-wang-dl** [comment]
@mengxr Run three times with Travis. And also multiple times on my local laptop. All tests are passed. The behavior seems fine.

**Comment by jkbradley** [comment]
For the record: Hossein found our NativeLibrary.java file was from TF master, not 1.14.0, and included other changes.  Double-checking this, I think this is a reasonable approach: The changes were relevant to the fix we need.  The changes were in the helper method used to identify the correct library and should not affect other parts of TensorFlow.  Here's a commit matching what we used for our patched file: https://github.com/tensorflow/tensorflow/blob/ed6fabc53d9732a4589eb617c5ba6f84f2b212a5/tensorflow/java/src/main/java/org/tensorflow/NativeLibrary.java

**Pull request merged**